src/osd/PrimaryLogPG.cc

```diff
@@ -506,6 +493,19 @@ void PrimaryLogPG::on_global_recover(
     waiting_for_unreadable_object.erase(unreadable_object_entry);
   }
   finish_degraded_object(soid);
+  // release obj rwlock after release degraded and unreadable_object, in some case will casuse out-of-order issue.
```

Comment:
Have you encountered an out-of-order bug due to this? If so, it will be good to add tracker issue with details.

Reply:
any more details on how this was found? did it cause an issue in a running system, or was it by code inspection?